client/src/App.tsx

```diff
@@ -1,38 +1,195 @@
+
+  const currentRoundRef = useRef<number>(0);
+  const currentPlayerRef = useRef<Player>(players[0]);
+
+  const showPlayerCard = (player: Player) => {
+    setGame((prevGame: Game) => {
+      return {
+        ...prevGame,
+        players: updatePlayerIsCardShown(prevGame.players, player.id),
+      };
+    });
+  };
+
+  const showAllCards = () => {
+    setGame((prevGame: Game) => {
+      return {
+        ...prevGame,
+        players: updatePlayerIsCardShownAll(prevGame.players, true),
+      };
+    });
+  };
+
+  const hideAllCards = () => {
+    setGame((prevGame: Game) => {
+      return {
+        ...prevGame,
+        players: updatePlayerIsCardShownAll(prevGame.players, false),
+      };
+    });
+  };
+
+  const nextPlayer = () => {
+    const prevPlayer = currentPlayerRef.current;
+    const prevPlayerIndex = players.findIndex(
+      (player) => player.id === prevPlayer.id
+    );
+    let nextPlayer: Player;
+    if (prevPlayerIndex === players.length - 1) {
+      nextPlayer = players[0];
+    } else {
+      nextPlayer = players[prevPlayerIndex + 1];
+    }
+    currentPlayerRef.current = nextPlayer;
+    startRound(nextPlayer);
+  };
+
+  const startRound = (player: Player) => {
+    currentRoundRef.current = currentRoundRef.current + 1;
+
+    setGame((prevGame: Game) => {
+      return {
+        ...prevGame,
+        currentRound: prevGame.currentRound + 1,
+      };
+    });
+
+    showPlayerCard(player);
+
+    if (!player.isHuman) {
+      const stat = chooseRandomStat(
+        player.cards[0].stats,
+        player.cards[0].stats.length
+      );
+      setTimeout(() => {
+        playRound(stat, player);
+      }, DEFAULT_TIMEOUT);
+    }
+  };
+
+  const playRound = (stat: Stat, player: Player) => {
+    showAllCards();
+
+    setTimeout(() => {
+      const { id } = calculateRoundWinner(players, stat, player);
+      console.log("winner " + id);
+
+      setGame((prevGame: Game) => {
+        return {
+          ...prevGame,
+          players: updatePlayerScores(prevGame.players, id),
+          roundWinners: updateRoundWinners(prevGame.roundWinners, id),
+          //roundsPlayed: prevGame.roundsPlayed + 1,
+        };
+      });
+
+      endRound();
+    }, DEFAULT_TIMEOUT);
+  };
+
+  const endRound = () => {
+    hideAllCards();
+
+    setTimeout(() => {
+      setGame((prevGame: Game) => {
+        return { ...prevGame, players: updatePlayerCards(prevGame.players) };
+      });
+      if (currentRoundRef.current === totalRounds) {
+        console.log("end of game");
+      } else {
+        nextPlayer();
+      }
+    }, DEFAULT_TIMEOUT);
+  };
+
   return (
     <>
+      <Header />
       <PlayerScore
         playerName={player1.name}
         playerId={player1.id}
         updateName={() => {}}
         playerScore={player1.score}
-        currentRound={game.currentRound}
-        totalRounds={game.totalRounds}
-        roundWinners={[]}
+        currentRound={currentRoundRef.current}
```

Comment:
I think we can get rid of currentRound in this component as roundWinners is passed through and we can use roundWinners.length instead in the child components. I can do a PR to show this.

Reply:
https://github.com/TeamTopTrumps/top-trumps-game/pull/25

Reply:
Re: discussion on Slack, currentRound is not always in sync with roundWinners.length. When a round starts currentRound is +1 ahead of roundWinners.length and this is reflected on the roundTracker (with a pokeball now I've done some styles!)

If you really don't want currentRound in state though then we could possible add a boolean for isGameActive or something? 
- When the game is initialised it would be false, so both roundWinners.length would be 0 and the current round would be 0.
- When the player starts the game and the first round, isGameActive could "activate" the roundTracker at round 1, so it knows current round is round 1
- After that the next current round would always be roundWinners.length + 1 until roundWinners.length === totalRounds and the game ends

Reply:
Okay happy to keep currentRound in. I'll close my PR.